src/pep8ext_naming.py

```diff
@@ -291,6 +291,25 @@ class ClassNameCheck(BaseASTCheck):
     Classes for internal use have a leading underscore in addition.
     """
     N801 = "class name '{name}' should use CapWords convention"
+    N818 = "class name '{name}' should use Error suffix as it is an Exception"
+
+    def get_class_def(self, name, parents):
+        for parent in parents:
+            for definition in parent.body:
+                class_definition = isinstance(definition, ast.ClassDef)
+                if class_definition and definition.name == name:
+                    return definition
+
+    def superclass_names(self, name, parents):
+        class_def = self.get_class_def(name, parents)
+        if not class_def:
+            return []
+        class_ids = []
```

Comment:
This method could return a `set` instead. It's result doesn't need to be ordered, and it's only used for a containment check below in `visit_classdef`.